### ISSUE TITLE
Facet search not applying `flimit` param and returns complete list of facets

### ISSUE THREAD

**Issue by qifeng-bai** [AtlasOfLivingAustralia/la-pipelines#437]
Comparing with the same query in biocache-ws-test and biocache prod,  I found

https://biocache-ws.ala.org.au/ws/occurrence/facets?facets=taxon_name&flimit=10&foffset=0&fsort=count&q=qid:1621139252199

Prod returns as JSON 

However, https://biocache-ws-test.ala.org.au/ws/occurrence/facets?facets=scientificName&flimit=10&foffset=0&fsort=count&q=qid:1624319447674

~returns a html page~

@alexhuang091 pointed out the test service is returning all results in the `fieldResult` JSON array, instead of the first 10 or the value provided by `flimit`.

**Comment by nickdos** [comment]
I'm not seeing HTML in the page source but the browser tool to format the JSON as pretty output is not being run on the test site but is on the prod site. The response content-type is application/json for both requests, so can't see why one is different to the other.

But the test version is returning 100x more data than the test site, so looks like something is amiss.

**Comment by alexhuang091** [comment]
<img width="390" alt="截屏2021-06-22 下午1 38 50" src="https://user-images.githubusercontent.com/61677987/122859444-6385d080-d35f-11eb-809d-7302d7b731e7.png">

Actually the return value is a List so will be converted to JSON, the thing is there are 187460 items in the list!

```
    public @ResponseBody
    List<FacetResultDTO> getOccurrenceFacetDetails(
```

**Comment by nickdos** [comment]
Agree @alexhuang091 seems the `flimit` is not being applied, maybe.

**Comment by qifeng-bai** [comment]
Agree @alexhuang091  @nickdos  It returns all recordes intead of 10 records limited by flimit

**Comment by alexhuang091** [comment]
I see, when I debugged, the flimit seems to be `-1`

**Comment by alexhuang091** [comment]
So the facet.limit controls how many returned?

<img width="282" alt="截屏2021-06-22 下午1 53 31" src="https://user-images.githubusercontent.com/61677987/122860559-4a7e1f00-d361-11eb-8c24-d153c3052621.png">

**Comment by nickdos** [comment]
`facet.limit` is not exposed as a public param for that service, so I think it should be mapped to the `flimit` param.

![image](https://user-images.githubusercontent.com/532845/122861429-caf14f80-d362-11eb-8e60-e38fc360aea0.png)

**Comment by alexhuang091** [comment]
`getFacetCounts` method on both pipeline and develop branch will explicitly call
`searchParams.setFlimit(-1);`

I'll keep debugging. It may take a while.

**Comment by javier-molina** [comment]
Nice pick @qifeng-bai is there a typical scenario to reproduce this using the UI? is so could you please add the steps to reproduce in the description? That will help gauge how critical this is.

I tried choose more... on biocache hub facets for taxon and apparently  Refine your search works fast and loading more records on demand so I don't want to make up scenarios. 

![Screen Shot 2021-06-22 at 2 43 29 pm](https://user-images.githubusercontent.com/8326394/122864579-3d186300-d368-11eb-9356-8e2140794e9f.png)

**Comment by adam-collins** [comment]
https://github.com/AtlasOfLivingAustralia/biocache-service/pull/646

**Comment by djtfmartin** [comment]
Fixed in 3.0.16

**Comment by nickdos** [comment]
LGTM ✅
